services/common/src/components/projectSummary/AuthorizationSupportDocumentUpload.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { FC } from "react";
+import { Field, getFormValues } from "redux-form";
+import { DOCUMENT, EXCEL, IMAGE, SPATIAL } from "@mds/common/constants/fileTypes";
+import { MINE_REPORT_DOCUMENT, PROJECT_SUMMARY_DOCUMENTS } from "@mds/common/constants/API";
+import RenderFileUpload from "../forms/RenderFileUpload";
+import { required, requiredNewFiles } from "@mds/common/redux/utils/Validate";
+import { IProjectSummaryDocument, IMineDocument } from "../..";
+import { useSelector } from "react-redux";
+import { FORM } from "../..";
+import { PROJECT_SUMMARY_DOCUMENT_TYPE_CODE } from "../..";
+
+interface AuthorizationSupportDocumentUploadProps {
+  mineGuid: string;
+  isProponent: boolean;
+  documents: IProjectSummaryDocument[];
+  updateAmendmentDocuments: (documents: IProjectSummaryDocument) => void;
+  projectGuid: string;
+  projectSummaryGuid: string;
+  dfaRequired: boolean;
+}
+
+export const AuthorizationSupportDocumentUpload: FC<AuthorizationSupportDocumentUploadProps> = ({
+  mineGuid,
+  isProponent,
+  documents,
+  updateAmendmentDocuments,
+  projectGuid,
+  projectSummaryGuid,
+  dfaRequired,
+}) => {
+  const handleRemoveFile = (error, fileToRemove) => {
+    if (error) {
+      console.log(error);
+    }
+    const newDocuments = documents.filter(
+      (file) => fileToRemove.serverId !== file.document_manager_guid
+    );
+    newDocuments.forEach((newDoc) => {
+      updateAmendmentDocuments(newDoc);
+    });
+  };
+
+  const handleFileLoad = (
+    document_name,
```

Comment:
Small thing, but it would be nice to have these props typed.

ie:

```
document_name: string,
    document_manager_guid: string,
    project_summary_document_type_code: string
```

Reply:
> Small thing, but it would be nice to have these props typed.
> 
> ie:
> 
> ```
>     document_name: string,
>     document_manager_guid: string,
>     project_summary_document_type_code: string
> ```

Updated accordingly.